Program.cs

```diff
@@ -3,17 +3,132 @@
 // Copyright (c) Metamation India.                                                
 // ---------------------------------------------------------------------------------------
 // Program.cs                                                                     
-// 
-// 
-// 
+// Program to solve the famous eight queens 
+// On a standard chessboard 8 queens have to be placed so that no queen can attack any other.
+// No two queens can be on the same row or column or diagonal.
+// Part - 1: Find all possible solutions.
+// Part - 2: Eliminate the identical solutions.
+// A solution is identical to another if it is a 90, 180 or 270 degree rotation of the other
+// or a vertical mirror image of the other.
 // ---------------------------------------------------------------------------------------
+using System.Text;
+using static System.Console;
+
 namespace Training {
    #region class Program --------------------------------------------------------------------------
+   /// <summary>Eight Queens problem</summary>
    internal class Program {
       #region Method ------------------------------------------------
-      static void Main (string[] args) {
-         Console.WriteLine ("Hello, World!");
+      /// <summary>This method prints all the required solutions</summary>
+      static void Main () {
+         WriteLine ("Press U to print unique solutions.\n" +
+                    "Else print any other key to print all solutions.");
+         bool isUnique = false;

```

Comment:
Can be simplified into 'PlaceQueen (Console.ReadKey (true).Key is ConsoleKey.U, 0);'

Reply:
This is a recursive method (line 118). So replacing it this way would ask the user multiple times to enter the same input.

Reply:
I think it'll not be. Let me know if you've further doubts.